src/cargo/ops/cargo_rustc/fingerprint.rs

```diff
@@ -148,11 +148,10 @@ impl Fingerprint {
     fn update_local(&self) -> CargoResult<()> {
         match self.local {
             LocalFingerprint::MtimeBased(ref slot, ref path) => {
-                let meta = fs::metadata(path).chain_error(|| {
-                    internal(format!("failed to stat `{}`", path.display()))
-                })?;
-                let mtime = FileTime::from_last_modification_time(&meta);
-                *slot.0.lock().unwrap() = Some(mtime);
+                if let Ok(meta) = fs::metadata(path) {
+                    let mtime = FileTime::from_last_modification_time(&meta);
+                    *slot.0.lock().unwrap() = Some(mtime);
+                }
```

Comment:
I think because with the RLS, we don't actually produce the file at `path`, and so we would get an error (even though we return `Stop`, there was no way to skip this code.